### PR TITLE
WIP Support protobuf, benchmarks and CM size analysis

### PR DESCRIPTION
**Pull request by pierDipi**
## Proposed Changes

- Support protobuf, benchmarks and CM size analysis

### CODE REVIEW

File: data-plane/tests/src/main/java/dev/knative/eventing/kafka/broker/tests/core/ConfigUnmarshalling.java [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+import joptsimple.internal.Strings;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.Fork;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.Warmup;
+
+@Fork(value = 1, warmups = 1)
```
Comment (slinkydeveloper) [inline data-plane/tests/src/main/java/dev/knative/eventing/kafka/broker/tests/core/ConfigUnmarshalling.java:27]: at least 5 warmups are needed

File: data-plane/tests/src/main/java/dev/knative/eventing/kafka/broker/tests/core/ConfigUnmarshalling.java [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+public class ConfigUnmarshalling {
+
+  private static final Brokers noBrokers = Brokers.newBuilder().build();
+  private static final Brokers full;
+  private static final int LONGEST_NAME = 253;
+  private static final int LONGEST_BROKER_NAME = LONGEST_NAME;
+  private static final int LONGEST_NAMESPACE_NAME = LONGEST_NAME;
+  private static final String K8S_SERVICE_SUFFIX = "svc.cluster.local"; // usual case
+
+  static {
```
Comment (slinkydeveloper) [inline data-plane/tests/src/main/java/dev/knative/eventing/kafka/broker/tests/core/ConfigUnmarshalling.java:37]: This static block, together with the alloc of `noBrokers`, should live in a state. If you alloc in a static, the jvm might perform some fancy optimizations on it

Comment (slinkydeveloper) [reply data-plane/tests/src/main/java/dev/knative/eventing/kafka/broker/tests/core/ConfigUnmarshalling.java:37]: What about encapsulating both in a static method and call this static method in the constructor of state?

**Review by slinkydeveloper — CHANGES_REQUESTED** [review]

**Comment by pierDipi** [comment]
Too many conflicts and too many changes happened as of late.
/close

**Pull request closed without merging**